src/Refactoring-UI/RePushUpMethodDriver.class.st

```diff
@@ -0,0 +1,241 @@
+"
+I represent a driver that invokes `PullUpMethod` refactoring.
+
+I am responsible for asking the user to which classes the method will be pulled up.
+
+When I gather all needed information I am calling and executing pull up method refactoring.
+
+You can create my instance and execute the refactoring by running:
+
+```
+(RBPushUpMethodDriver model: aRBNamespace scopes: refactoringScopes pushDown: methods) runRefactoring
+```
+"
+Class {
+	#name : 'RePushUpMethodDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'methods',
+		'class',
+		'methodSelectionPresenterClass',
+		'superclass',
+		'notOverrides',
+		'notInstVarRefs',
+		'notSharedVarRefs',
+		'notSupersendsReceived',
+		'notSupersendsSent'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'initialization' }
+RePushUpMethodDriver class >> model: model scopes: refactoringScopes pushUpMethods: methods [
+
+	^ self new
+		  model: model
+		  scopes: refactoringScopes
+		  pushUpMethods: methods
+]
+
+{ #category : 'ui - dialogs' }
+RePushUpMethodDriver >> alertFailedPreconditions [
+	SpAlertDialog new
+		title: 'Push Up failed: Required preconditions were not met';
+		label:  self failedPreconditionsErrorString;
+		acceptLabel: 'Accept';
+		openDialog
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> breakingChoices [
+
+	^ refactoring failedBreakingChangePreconditions collect: [ :each |
+		each class strategyChoiceClass new driver: self ]
+]
```

Comment:
This is cool! However, we should support multiple choices per breaking change preconditions. This can be dealt with in future PR, not this one.

Reply:
Yes in another step :)

Reply:
Ok, I'll keep it in mind